services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/device/AbstractRegistrationService.java

```diff
@@ -194,7 +351,42 @@ public final void setTenantInformationService(final TenantInformationService ten
                                 final RegistrationResult deviceResult = deviceInfoTracker.result();
                                 final RegistrationResult gatewayResult = gatewayInfoTracker.result();
 
-                                if (!isDeviceEnabled(deviceResult)) {
+                                if (deviceResult.isNotFound() && !gatewayResult.isNotFound()
+                                        && isDeviceEnabled(gatewayResult) && hasAuthorityForAutoRegistration(gatewayResult)) {
+
+                                    final Device device = new Device()
+                                        .setEnabled(true)
+                                        .setAutoProvisioned(true)
+                                        .setVia(Collections.singletonList(gatewayId));
+
+                                    final JsonArray memberOf = gatewayResult.getPayload().getJsonObject(RegistrationConstants.FIELD_DATA)
+                                            .getJsonArray(RegistryManagementConstants.FIELD_MEMBER_OF);
+                                    if (memberOf != null && !memberOf.isEmpty()) {
+                                        device.setViaGroups(memberOf.getList());
+                                    }
+
+                                    LOG.debug("auto-provisioning device {} for gateway {}", deviceId, gatewayId);
```

Comment:
I would be good to create a new child span here, and use it in this block (for everything except `createSuccessfulRegistrationResult`, I guess). That would allow filtering for that operation in the Tracing UI.

Reply:
That makes sense. But I would also use the child span for `createSuccessfulRegistrationResult` since it is also done in the context of auto provisioning.

Reply:
I just think the auto provisioning is done at that point, and a possible error like "failed to resolve group members", returned in `createSuccessfulRegistrationResult`, would be better of getting logged in the `assert Device Registration` span.